download_lenta.py

```diff
@@ -137,8 +175,10 @@ async def _fetch_all_news_on_page(self, html: str):
             except Exception:
                 logger.exception(f"Cannot parse {url}")
             else:
-                parse_res["url"] = url
-                parsed_news.append(parse_res)
+                if url not in self._existing_urls:
```

Comment:
It's better to make `self._existing_urls` as `dict` instead of `list` to speed up the search here.

Reply:
agree, miss it. Updated